Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2024 Colin Jones. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Colin Jones
+-/
+import Mathlib.Algebra.GeomSum
+import Mathlib.NumberTheory.Divisors
+import Mathlib.Tactic.FinCases
+import Mathlib.Tactic.NormNum.Prime
+
+/-!
+# Factorisation properties of natural numbers
+
+This file defines abundant, pseudoperfect, deficient, and weird numbers and formalizes their
+relations with prime and perfect numbers.
+
+## Main Definitions
+
+* `Nat.Abundant`: a natural number `n` is _abundant_ if the sum of its proper divisors is greater
+  than `n`
+* `Nat.Pseudoperfect`: a natural number `n` is _pseudoperfect_ if the sum of a subset of its proper
+  divisors equals `n`
+* `Nat.Deficient`: a natural number `n` is _deficient_ if the sum of its proper divisors is less
+  than `n`
+* `Nat.Weird`: a natural number is _weird_ if it is abundant but not pseudoperfect
+
+## Main Results
+
+* `Nat.deficient_or_perfect_or_abundant`: A positive natural number is either deficient,
+  perfect, or abundant.
+* `Nat.prime_deficient`: All prime natural numbers are deficient.
+* `Nat.exists_infinite_deficients`: There are infinitely many deficient numbers.
+* `Nat.prime_pow_deficient`: Any natural number power of a prime is deficient.
```

Comment:
Those lemma names are outdated

Reply:
Should I keep the Nat in front of infinite_deficient?

Reply:
Yes, why not? Its full name is `Nat.Prime.pow_deficient`, right?